File: Implementation - C++/Dog.cpp
#include "Dog.h"
#include <set>
#include <iostream>

using namespace std;

/**
 * Dog: an implementation class for storing the details of dogs.
 *
 * Written for COMP8741 Application Development - Research Project
 * by Joel Pillar-Rogers
 * on 4 June 2019
 */


/**
* Creates a dog
*/
Dog::Dog(string name, string dogBreed, int regNum)
{
    name_ = name;
    dogBreed_ = dogBreed;
    regNum_ = regNum;
    dogCount_++;
    seqNum_ = dogCount_;
}

/**
* Getters and Setters
*/
string Dog::getName()
{
    return name_;
}

void Dog::setName(string name)
{
    name_ = name;
}

string Dog::getDogBreed()
{
    return dogBreed_;
}

void Dog::setDogBreed(string dogBreed)
{
    dogBreed_ = dogBreed;
}

int Dog::getRegNum()
{
    return regNum_;
}

void Dog::setRegNum(int regNum)
{
    regNum_ = regNum;
}

set<string> Dog::getOwners()
{
    return owners_;
}

void Dog::addOwner(string newOwner)
{
    owners_.insert(newOwner);
}

/**
* Prints the total number of dogs bred
*/
void Dog::printTotal()
{
    cout << "Total number of dogs bred = " << dogCount_ << endl;
}
/**
* Prints all the details of this dog
*/
void Dog::print()
{
    cout << "Dog " << name_ << ", Breed = " << dogBreed_ << ", #owners = " << owners_.size() << ", Registration # = " << regNum_ << ", Sequence # = " << seqNum_ << endl;
}
/**
 * definition of the dogCount static variable
 */
int Dog::dogCount_;

File: Implementation - Java/src/practical2/Dog.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package practical2;

import java.util.HashSet;
import java.util.Set;

/**
 *
 * @author Joel
 */
public class Dog {

    private String name = "unknown";
    private String dogBreed = "unknown";
    private int regNum = -1;
    private final Set<String> owners = new HashSet<>();
    private static int dogCount = 0;
    private int seqNum = 0;

    public Dog() {
        dogCount++;
        seqNum = dogCount;
    }

    public Dog(String name, String breed, int regNum) {
        this.name = name;
        this.dogBreed = breed;
        this.regNum = regNum;
        dogCount++;
        seqNum = dogCount;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getDogBreed() {
        return dogBreed;
    }

    public void setDogBreed(String dogBreed) {
        this.dogBreed = dogBreed;
    }

    public int getRegNum() {
        return regNum;
    }

    public void setRegNum(int regNum) {
        this.regNum = regNum;
    }

    public Set<String> getOwners() {
        return owners;
    }

    public int getNumberOwners() {
        return owners.size();
    }

    public void addOwner(String newOwner) {
        owners.add(newOwner);
    }

    public static void printTotal() {
        System.out.println("Total number of dogs bred = " + dogCount);
    }

    public String toString() {
        return "Dog " + name + ", Breed = " + dogBreed + ", #owners = " + owners.size() + ", Registration # = " + regNum + ", Sequence # = " + seqNum;
    }

}
